### ISSUE TITLE
CN hash algorithms not working in debug builds

### ISSUE THREAD

**Issue by CodeIsTheKey** [Raptor3um/Raptoreum#126]
When building with --enable-debug, the CN algorithms are not computing the same has as the normal release build.  The core hashing algorithms are working fine, but the hash computation goes awry on the first CN hashing.

While doing some debugging, I noticed for CNDark and CNTurtlelite did not behave as expected.  I believe the problem covers all of the CN algorithms.

**Comment by malbit** [comment]
> When building with --enable-debug, the CN algorithms are not computing the same has as the normal release build. The core hashing algorithms are working fine, but the hash computation goes awry on the first CN hashing.
> 
> While doing some debugging, I noticed for CNDark and CNTurtlelite did not behave as expected. I believe the problem covers all of the CN algorithms.

Would you be able to provide some logs or more information? What part not behaving as expected and what you mean by "not behave" ??

**Comment by CodeIsTheKey** [comment]
The hashing within HashGR seems to be going off, causing the CheckProofOfWork to fail.

At the end of HashGR, my code with logging looks like:
```
        std::cout << "HashGR, i: " << i << ", coreSelection: " << coreSelection << ", cnSelection: " << cnSelection << ", hash: " << std::endl;
        coreHash(toHash, &hash[i], lenToHash, coreSelection);
        std::cout << "   after coreHash: " << hash[i].ToString() << std::endl;
        cnHash(&hash[i - 1], &hash[i], lenToHash, cnSelection);
        std::cout << "   after cnHash: " << hash[i].ToString() << std::endl;
```

In a release build, the final POW hash looks good, meets ProofOfWork contraints, and matches the chain.
[release-hash.log](https://github.com/Raptor3um/raptoreum/files/8198181/release-hash.log)

Doing a full (clean) rebuild after running ``./configure --prefix=`pwd`/depends/x86_64-linux-gnu/ --enable-debug`` I get the following:
[debug-hash.log](https://github.com/Raptor3um/raptoreum/files/8198211/debug-hash.log)

The results are identical in the two logs until we reach the first CN hashing.  The problem is not specific to the CN algorithm select - my logs look the same for multiple tests when different CN algorithms are selected.

My first assumption was that there is some alignment issue in the debug vs release build that is causing the problem.  I looked, but didn't see anything obvious.

I have tested this on the current develop branch as well as master and the results are consistent - release builds work and debug builds do not.
